core/src/test/java/org/pentaho/platform/util/EmailerTest.java

```diff
@@ -82,4 +122,153 @@ public void setBcc_ReplaceToCommaTest() {
+    final String body = UUID.randomUUID().toString();
+    emailer.setBody( body );
+    emailer.send();
+    assertEquals( 1, MockMail.size() );
+    final Message message = MockMail.get( 0 );
+    assertNotNull( message );
+    final MimeMultipart content = (MimeMultipart) message.getContent();
+    assertNotNull( content );
+    assertEquals( 1, content.getCount() );
+    final BodyPart bodyPart = content.getBodyPart( 0 );
+    assertNotNull( bodyPart );
+    assertEquals( "text/plain; charset=UTF-8", bodyPart.getContentType() );
+    final String bodyPartContent = (String) bodyPart.getContent();
+    assertEquals( body, bodyPartContent );
+  }
+
+  @Test
+  public void testEmbeddedHtmlBodyWithAttachment() throws IOException, MessagingException {
+    emailer.setTo( "tets@test.email.com" );
+    emailer.setFrom( "tets@test.email.com" );
+    emailer.setSubject( "Test" );
+    final String body = UUID.randomUUID().toString();
+    emailer.setBody( body );
+    emailer.setAttachmentMimeType( "text/html" );
+    try ( ByteArrayInputStream stream = new ByteArrayInputStream( eml.getBytes() ) ) {
+      emailer.setAttachment( stream );
+      emailer.setAttachmentName( "email-file" );
+    }
+    emailer.send();
+    assertEquals( 1, MockMail.size() );
+    final Message message = MockMail.get( 0 );
+    assertNotNull( message );
+    final MimeMultipart content = (MimeMultipart) message.getContent();
+    assertNotNull( content );
+    assertEquals( 2, content.getCount() );
+    final BodyPart bodyPart = content.getBodyPart( 0 );
+    assertNotNull( bodyPart );
+    assertEquals( "text/plain; charset=UTF-8", bodyPart.getContentType() );
+    final String bodyPartContent = (String) bodyPart.getContent();
+    assertEquals( body, bodyPartContent );
+  }
+
+  @Test
+  public void testGetSmtpSessionOAuth2() {
+    emailer.setSmtpHost( "smtp.outlook.com" );
+    emailer.setSmtpPort( 587 );
+    emailer.setTransportProtocol( "SMTP" );
+    emailer.setStartTLS( true );
+    emailer.setUseSSL( true );
+    emailer.setQuitWait( false );
+    emailer.setFromName( "test@pentaho.com" );
+    emailer.setUseAuthentication( true );
+    emailer.setAuthMechanism( "XOAUTH2" );
+    assertEquals( "587", emailer.getSmtpSession( true ).getProperty( "mail.smtp.port" ) );
+  }
+
+  @Test
+  public void testSend() {
+    emailer.setSmtpHost( "smtp.outlook.com" );
+    emailer.setSmtpPort( 587 );
+    emailer.setTransportProtocol( "SMTP" );
+    emailer.setStartTLS( true );
+    emailer.setUseSSL( true );
+    emailer.setQuitWait( false );
+    emailer.setFromName( "test@pentaho.com" );
+    emailer.setUseAuthentication( true );
+    emailer.setAuthMechanism( "basic" );
+    emailer.getSmtpSession( true );
+    emailer.setTo( "tets@test.email.com" );
+    emailer.setFrom( "tets@test.email.com" );
+    emailer.setSubject( "Test" );
+    final String body = UUID.randomUUID().toString();
+    emailer.setBody( body );
+    emailer.setAuthenticator( new Authenticator() {
```

Comment:
for ` emailer.setAuthenticator( ...`
You aren't using testing the Authenicator logic, just use a mock.